### PR TITLE
Migrate Tenants screen to Angular

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

https://github.com/gravitee-io/issues/issues/6369

**Description**

 - Migrate Tenants screen and service to Angular
 
**Screenshot**

Before: 
image

After: 
image

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-cvdsrdimrj.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6369-migrate-tenant-screen/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-hqkipvhcwf.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6369-migrate-tenant-screen/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html [hunk 1]
```diff
@@ -15,4 +15,74 @@
...
+</h4>
+
+<div class="add-button-container" *gioPermission="{ anyOf: ['organization-tenant-c'] }">
+  <button (click)="onAddTenantClicked()" mat-raised-button color="primary"><mat-icon>add</mat-icon> Add a tenant</button>
+</div>
+
+<div class="mat-elevation-z3">
+  <gio-table-wrapper (filtersChange)="onFiltersChanged($event)">
+    <table mat-table [dataSource]="tenantsDataSource" class="tenants-table" id="tenantsTable" aria-label="Tenant table">
+      <!-- Email Column -->
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html]: Id Column

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html [hunk 2]
```diff
@@ -15,4 +15,74 @@
...
+      <ng-container matColumnDef="id">
+        <th mat-header-cell *matHeaderCellDef id="id">Id</th>
+        <td mat-cell *matCellDef="let element">{{ element.id }}</td>
+      </ng-container>
+
+      <!-- Email Column -->
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html]: Name Column

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html [hunk 3]
```diff
@@ -15,4 +15,74 @@
...
+      <ng-container matColumnDef="name">
+        <th mat-header-cell *matHeaderCellDef id="name">Name</th>
+        <td mat-cell *matCellDef="let element">{{ element.name }}</td>
+      </ng-container>
+
+      <!-- Email Column -->
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html]: Description :p

Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.html]: :aa:

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.scss [hunk 4]
```diff
@@ -3,3 +3,35 @@
...
+  padding: 0 4px;
+}
+
+.centered-cell {
+  display: flex;
+  justify-content: center;
+}
+
+.add-button-container {
+  width: 100%;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.scss]: I always wonder about using  `width` or `flex:...`

https://stackoverflow.com/a/34355447
🤷‍♂️

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-add-tenant.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+import { OrganizationSettingsModule } from '../organization-settings.module';
+import { fakeTenant } from '../../../entities/tenant/tenant.fixture';
+
+describe('GioConfirmDialogComponent', () => {
+  let component: OrgSettingAddTenantComponent;
+  let fixture: ComponentFixture<OrgSettingAddTenantComponent>;
+  let loader: HarnessLoader;
+
+  const matDialogRefMock = {
+    close: jest.fn(),
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-add-tenant.component.spec.ts:34]: maybe add jest reset mock in after/ before each ?

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.ts [hunk 6]
```diff
@@ -82,8 +83,27 @@ export class OrgSettingsTenantsComponent implements OnInit, OnDestroy {
...
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function,@typescript-eslint/no-unused-vars
-  onDeleteTenantClicked(element: any) {}
+  onDeleteTenantClicked(tenant: Tenant) {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        width: '450px',
+        data: {
+          title: 'Delete a tenant',
+          content: `Are you sure you want to remove the tenant <strong>${tenant.name}</strong>?`,
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.ts:92]: I always wonder if we should add or not a space before `?` 🤔 

What is your opinion on this?

File: gravitee-apim-console-webui/src/shared/components/gio-table-wrapper/gio-table-wrapper.util.ts [hunk 7]
```diff
@@ -43,11 +43,8 @@ export const gioTableFilterCollection = <T>(
   }
 
   if (filters?.sort) {
-    sortedCollection = sortBy(sortedCollection, filters.sort.active);
-
-    if (filters.sort.direction === 'desc') {
-      sortedCollection = sortedCollection.reverse();
-    }
+    const sortDirection = filters.sort.direction === '' ? 'asc' : filters.sort.direction;
+    sortedCollection = orderBy(sortedCollection, filters.sort.active, sortDirection);
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/shared/components/gio-table-wrapper/gio-table-wrapper.util.ts:47]: mazette 👌

**Review by ThibaudAV — APPROVED** [review]

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.ts (hunk 6 above)
Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-tenants.component.ts:92]: No space in 🇬🇧  (but one in 🇫🇷  🙃 )

> Question marks [?]
--
Question marks go at the end of sentences that are questions. As with full stops you do not put a space before a question mark, but you do need at least one space after one (two spaces for purists).
For example:-Is my name Lynne? Of course it is.

File: gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-add-tenant.component.spec.ts (hunk 5 above)
Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-add-tenant.component.spec.ts:34]: 💯

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/tenants/org-settings-add-tenant.component.spec.ts:34]: we could also add this in 
setup-jest  🤔

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-qtdxmzwlit.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6369-migrate-tenant-screen/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Pull request merged**